wordpress-mobile/WordPress-Android
# [JP Social/Pre-publish] Integrate real data in the pre-publish home bottom sheet

Part of #18876 
Needs to be merged only after #18883

## Note to reviewers / testers
You might notice the UI in the Post Settings looks a bit odd, but that's because the UI fixes there are not yet merged. These fixes are available here: https://github.com/wordpress-mobile/WordPress-Android/pull/18884

## To test
Ideally, you need several sites to test the several different scenarios, such as:
- WPCOM site with at least 1 social connection (no share limit)
- Self-hosted with JP Connected and JP Social installed and at least 1 social connection (share limits)
- Site with JP Social but no social connections

Then you can do these steps with each of them:
1. Open app
2. Enable the `FeatureInDevelopment` for Jetpack Social in Debug Settings
3. Select the site
4. Go to `Posts`
5. Create a new post
6. Go to the `Post Settings` and check what the `Social Sharing` section looks like
7. Hit `publish`
8. **Verify** the Social item matches the `Post Settings` state
9. Go back and change some `Social Sharing` options (if possible)
10. Hit `publish`
11. **Verify** the Social item matches the `Post Settings` state again

Try changing the settings to achieve several states (sharing to 0, 1, 2+ networks). Also make sure that hitting `not now` button for the "no connections" UI (when the site has no social connections) only affects the visibility of that flow, e.g.: hiding the item in the "pre-publish" sheet should not hide the section inside the post settings and vice-versa.

## Regression Notes
1. Potential unintended areas of impact
Broken state in Post Settings.

13. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

14. What automated tests I added (or what prevented me from doing so)
Updated tests.

PR submission checklist:


UI Changes testing checklist:
